rpc/src/evm_rpc_impl/mod.rs

```
@@ -104,6 +105,7 @@ impl StateRootWithBank {
 
 #[instrument(skip(meta))]
 async fn block_to_state_root(
+    chain_id: ChainID,
```

This method should return an error in case when chain id is invalid.

Please convert it result type to 

`Result<StateRootWithBank, Error>`

And add case for `Error` where chain id is not found (for now should be returned when `chain_id != meta.get_main_chain_id()` )